packages/block-editor/src/components/global-styles/typography-utils.js

```diff
@@ -85,3 +76,36 @@ export function getTypographyFontSizeValue( preset, typographySettings ) {
 
 	return defaultSize;
 }
+
+function isFluidTypographyEnabled( typographySettings ) {
+	const fluidSettings = typographySettings?.fluid;
+	return (
+		true === fluidSettings ||
+		( typeof fluidSettings === 'object' &&
+			Object.keys( fluidSettings ).length > 0 )
+	);
+}
+
+/**
+ * Returns fluid typography settings from theme.json setting object.
+ *
+ * @param {Object} settings            Theme.json settings
+ * @param {Object} settings.typography Theme.json typography settings
+ * @param {Object} settings.layout     Theme.json layout settings
+ * @return {{fluid: (*&{maxViewPortWidth})}|{fluid: *}} Fluid typography settings
```

Comment:
Nit: it looks like the right hand type is more permissive than the left one, so I'm wondering if the `*&{maxViewPortWidth})}` is necessary? In VSCode if I assign `let item = getFluidTypographyOptionsFromSettings( tree?settings );` the result appears to be `any` for `fluid` in all cases:

<img width="717" alt="image" src="https://github.com/WordPress/gutenberg/assets/14988353/672b9105-dcd7-495a-86e0-2e1c82a72b5c">

Reply:
Ugh, this was my IDE. Please believe me! 😄 Thanks for spotting it 🙇 

I'd already written a type object above `TypographySettings`. Auto format killed it!